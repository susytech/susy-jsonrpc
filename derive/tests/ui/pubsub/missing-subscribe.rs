#[macro_use]
extern crate susy_jsonrpc_derive;
extern crate susy_jsonrpc_core;
extern crate susy_jsonrpc_pubsub;

#[rpc]
pub trait Rpc {
	type Metadata;

	// note that a subscribe method is missing

	/// Unsubscribe from hello subscription.
	#[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]
	fn unsubscribe(&self, Option<Self::Metadata>, SubscriptionId) -> Result<bool>;
}

fn main() {}
