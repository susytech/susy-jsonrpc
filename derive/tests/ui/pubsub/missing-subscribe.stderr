error: custom attribute panicked
 --> $DIR/missing-subscribe.rs:6:1
  |
6 | #[rpc]
  | ^^^^^^
  |
  = help: message: [rpc] encountered error: subscription 'hello'. Can't find subscribe method, expected a method annotated with `subscribe` e.g. `#[pubsub(subscription = "hello", subscribe, name = "hello_subscribe")]`

error: aborting due to previous error

